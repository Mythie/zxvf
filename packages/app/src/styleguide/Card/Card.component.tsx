import { defineComponent, PropType, toRefs } from 'vue';

export interface CardComponentProps {
  title: string;
}

export const CardComponent = defineComponent({
  name: 'Card',

  props: {
    title: {
      type: String as PropType<CardComponentProps['title']>,
      default: '',
    },
  },

  setup: (props, { slots }) => {
    const { title } = toRefs(props);

    return () => (
      <div class="w-full rounded-md border border-gray-50 bg-white p-5 shadow dark:border-slate-500 dark:bg-slate-700 dark:text-white">
        {title && <h4 class="pb-3 text-lg font-medium leading-tight">{title.value}</h4>}

        <div class="leading-relaxed">{slots.default && slots.default()}</div>
      </div>
    );
  },
});
