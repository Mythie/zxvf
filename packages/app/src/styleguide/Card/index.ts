export { CardComponent as Card } from './Card.component';

export * from './Card.component';
