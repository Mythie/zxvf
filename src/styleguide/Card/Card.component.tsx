import { defineComponent, PropType, toRefs } from 'vue';

export interface CardComponentProps {
  title: string;
}

export const CardComponent = defineComponent({
  name: 'Card',

  props: {
    title: {
      type: String as PropType<CardComponentProps['title']>,
      default: '',
    },
  },

  setup: (props, { slots }) => {
    const { title } = toRefs(props);

    return () => (
      <div class="w-full p-5 bg-white border rounded-md shadow dark:bg-slate-700 dark:text-white border-gray-50 dark:border-slate-500">
        {title && <h4 class="pb-3 text-lg font-medium leading-tight">{title.value}</h4>}

        <div class="leading-relaxed">{slots.default && slots.default()}</div>
      </div>
    );
  },
});
