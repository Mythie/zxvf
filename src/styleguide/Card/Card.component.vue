<script lang="ts" setup>
import { PropType } from 'vue';

defineProps({
  title: {
    type: String as PropType<string>,
    default: '',
  },
});
</script>

<template>
  <div
    class="w-full p-5 bg-white border rounded-md shadow dark:bg-slate-700 dark:text-white border-gray-50 dark:border-slate-500"
  >
    <template v-if="title">
      <h4 class="pb-3 text-lg font-medium leading-tight">
        {{ title }}
      </h4>
    </template>

    <div class="leading-relaxed">
      <slot />
    </div>
  </div>
</template>
