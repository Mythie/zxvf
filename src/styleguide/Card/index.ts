export { default as Card } from './Card.component.vue';
