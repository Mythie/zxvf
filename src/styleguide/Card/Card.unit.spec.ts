import { mount } from '@vue/test-utils';
import { describe, expect, it } from 'vitest';

import Card from './Card.component.vue';

describe('Card.component.vue', () => {
  it('should render correctly', () => {
    const wrapper = mount(Card);

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render correctly with props', () => {
    const wrapper = mount(Card, {
      propsData: {
        title: 'My Title',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render correctly with props and slot', () => {
    const wrapper = mount(Card, {
      propsData: {
        title: 'My Title',
      },
      slots: {
        default: '<p>My content</p>',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
