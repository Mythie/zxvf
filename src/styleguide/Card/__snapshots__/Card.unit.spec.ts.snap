// Vitest Snapshot v1

exports[`Card.component.vue > should render correctly 1`] = `
<div
  class="w-full p-5 bg-white border rounded-md shadow dark:bg-slate-700 dark:text-white border-gray-50 dark:border-slate-500"
>
  <!--v-if-->
  <div
    class="leading-relaxed"
  >
    
    
  </div>
</div>
`;

exports[`Card.component.vue > should render correctly with props 1`] = `
<div
  class="w-full p-5 bg-white border rounded-md shadow dark:bg-slate-700 dark:text-white border-gray-50 dark:border-slate-500"
>
  <h4
    class="pb-3 text-lg font-medium leading-tight"
  >
    My Title
  </h4>
  <div
    class="leading-relaxed"
  >
    
    
  </div>
</div>
`;

exports[`Card.component.vue > should render correctly with props and slot 1`] = `
<div
  class="w-full p-5 bg-white border rounded-md shadow dark:bg-slate-700 dark:text-white border-gray-50 dark:border-slate-500"
>
  <h4
    class="pb-3 text-lg font-medium leading-tight"
  >
    My Title
  </h4>
  <div
    class="leading-relaxed"
  >
    
    <p>
      My content
    </p>
    
  </div>
</div>
`;
